Create kotlin.md (#1166)
Codecademy/docs: content/Kotlin/kotlin.md

---
Title: 'Kotlin'

> Description: 'Kotlin is a contemporary language released by JetBrains in 2011. It is becoming popular and quickly replacing Java as the choice for Android software projects.'

Reviewer: Ideally, the `Description` should be one sentence that describes what Kotlin is and is commonly used for.

```suggestion
Description: 'Kotlin is a programming language used for general purposes including front-end, back-end, and mobile development.'
```

> Description: 'Kotlin is a contemporary language released by JetBrains in 2011. Kotlin is becoming popular and is fast replacing Java as people’s choice for software projects.'

Reviewer 2: ```suggestion
Description: 'Kotlin is a contemporary language released by JetBrains in 2011. It is becoming popular and quickly replacing Java as the choice for Android software projects.'
```

Reviewer 2: I think it'd also be good to match the introductory paragraph to the suggestion above

Codecademy Hub Page: 'https://www.codecademy.com/catalog/language/kotlin'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> [Kotlin](https://www.codecademy.com/catalog/language/kotlin) is a contemporary language released by JetBrains in 2011. Kotlin is becoming popular and is fast replacing Java as people’s choice for software projects. Java and Kotlin are compatible and proficiency in Java makes learning Kotlin much easier. 

Reviewer: ```suggestion
[**Kotlin**](https://www.codecademy.com/catalog/language/kotlin) is a programming language used for general purposes including [front-end](https://www.codecademy.com/resources/docs/general/front-end), [back-end](https://www.codecademy.com/resources/docs/general/back-end), and mobile development. Since Kotlin code is fully compatible with Java, its syntax may seem familiar to those with proficiency in Java.

After being released by JetBrains in 2011, Kotlin became the preferred language for Android development by Google in 2019.
```

I'm not sure Docs is the right place to state whether one language is replacing another. However, I do think it is fair to say that having a background in Java makes working with Kotlin relatively easier. 😄

> ##Uses of Kotlin

Reviewer 2: ```suggestion
## Uses of Kotlin
```

> Back-end web development

Reviewer 3: This should be a bulleted list with an introductory sentence.

Author: @SSwiniarski noted. Thanks

> Back-end web development
> Android development
> Full-stack web development
> Data Science
> Multi-platform mobile development

Reviewer: To build on this, I would also suggest the following edits:

```suggestion
- Android development
- Back-end development
- Front-end development
- [Full-stack](https://www.codecademy.com/resources/docs/general/full-stack) development
- [Data Science](https://www.codecademy.com/resources/docs/general/data-science)
- Multi-platform mobile development
```
